src/net.c

```diff
@@ -1833,15 +1813,26 @@ handle_request(coap_context_t *context, coap_session_t *session, coap_pdu_t *pdu
   coap_option_filter_clear(opt_filter);
 
   /* try to find the resource from the request URI */
-  coap_hash_request_uri(pdu, key);
-  resource = coap_get_resource_from_key(context, key);
+  str *uri_path = coap_get_uri_path(pdu);
+  if (!uri_path)
+    return;
+  resource = coap_get_resource_from_uri_path(context, *uri_path);
 
-  if (!resource) {
+  if (!resource || resource->unknown) {
     /* The resource was not found. Check if the request URI happens to
-     * be the well-known URI. In that case, we generate a default
-     * response, otherwise, we return 4.04 */
-
-    if (is_wkc(key)) {	/* request for .well-known/core */
+     * be the well-known URI, a PUT or a DELETE.
+     *
+     * If well-known URI generate a default response
+     *
+     * Else If unknown URI handler defined, call the unknown
+     *  URI handler (to allow for potential generation of resource
+     *  [RFC7272 5.8.3]) or do other checks that generates a suitable response
+     *
+     * Else If DELETE return 2.02 (RFC7252: 5.8.4.  DELETE)
+     *
+     * Else return 4.04 */
+
+    if (!strcmp ((char*)uri_path->s, COAP_DEFAULT_URI_WELLKNOWN)) {	/* request for .well-known/core */
```

Comment:
Please do not ever use `strcmp()`. 

 `strncmp()` might work but is not available on all platform. I suggest using `memcmp()` instead.